src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -31,5 +48,41 @@ public abstract class CompilationModuleGroup
         /// accessed through the target platform's import mechanism (ie, Import Address Table on Windows)
         /// </summary>
         public abstract bool ShouldReferenceThroughImportTable(TypeDesc type);
+
+        private class CompilerGeneratedAssembly : ModuleDesc, IAssemblyDesc
```

Comment:
This feels like it should live somewhere more type-system related since it's pretending to be an assembly. Maybe it's time to organize the sources under ILCompiler.Compiler\Compiler and add \TypeSystem for these kinds of classes.